No info about ASYNCIO in dramatiq(documentation, dramatiq --help coomand, examples)

Need add full examples and info about ASYNCIO in dramatiq

> I'm happy to help with documentation, but I have to admit I'm confused as to how to get it to work. Getting the synchronous example from the homepage `example.py` works fine. I then made it async:
> 
> ```
> @dramatiq.actor
> async def count_words(url):
>     response = await requests.get(url)
>     count = len(response.text.split(" "))
>     print(f"There are {count} words at {url!r}.")
> 
> 
> if __name__ == "__main__":
>     count_words.send(sys.argv[1])
> ```
> 
> Rerunning this results in a RuntimeException:
> 
> ```
> [2024-05-29 12:39:03,058] [PID 57175] [Thread-7] [dramatiq.worker.WorkerThread] [ERROR] Failed to process message count_words('https://github.com') with unhandled exception.
> Traceback (most recent call last):
>   File "/Users/rdl/miniconda3/envs/bot/lib/python3.12/site-packages/dramatiq/worker.py", line 487, in process_message
>     res = actor(*message.args, **message.kwargs)
>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/rdl/miniconda3/envs/bot/lib/python3.12/site-packages/dramatiq/actor.py", line 182, in __call__
>     return self.fn(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/rdl/miniconda3/envs/bot/lib/python3.12/site-packages/dramatiq/asyncio.py", line 63, in wrapper
>     raise RuntimeError(
> RuntimeError: Global event loop thread not set. Have you added the AsyncIO middleware to your middleware stack?
> ```
> 
> I believe the proximate cause of this is because `before_worker_boot` in `middleware.asyncio.AsyncIO` is not being called, which means the `event_loop_thread` is not set. However, I don't understand enough of the internals as to why that's the case, as I see from the docs that AsyncIO middleware is enabled by default.

The aio middleware is not enabled by default. The list of default middleware is here: https://github.com/Bogdanp/dramatiq/blob/052fea034e0457394442ae4badfcbfb000a7d87a/dramatiq/middleware/__init__.py#L47

--- a/docs/source/advanced.rst
+++ b/docs/source/advanced.rst
@@ -3,6 +3,8 @@
 Advanced Topics
 ===============
 
+.. _customizing-middleware:
+
 Customizing Middleware
 ----------------------
 

--- a/docs/source/cookbook.rst
+++ b/docs/source/cookbook.rst
@@ -144,6 +144,37 @@ call |pipeline_get_results|::
   for res in pipe.get_results(block=True):
       ...
 
+Asyncio Actors
+--------------
+
+To use asynchronous functions (``async def``) as actors,
+you must enable the optional |AsyncIO| middleware (See :ref:`customizing-middleware`).
+
+.. code-block:: python
+
+   import asyncio
+   import dramatiq
+   import dramatiq.middleware
+
+   broker = dramatiq.Broker()
+   dramatiq.set_broker(broker)
+   broker.add_middleware(dramatiq.middleware.AsyncIO())
+
+   @dramatiq.actor
+   async def waiter():
+       await asyncio.sleep(10)
+
+This middleware will start up 1 thread per worker process to run an event loop,
+where any asynchronous actor functions will be scheduled to run.
+
+You can use a mixture of synchronous actors and asynchronous actors,
+the synchronous ones will be run on the worker threads as normal,
+and the asynchronous ones will be scheduled on the event loop thread.
+
+Note that the maximum number of concurrent messages processed by a worker process,
+is still equal to the number of worker threads (8 threads by default).
+This is because each worker thread waits for the result of an asynchronous actor
+(same as a synchronous actor) before starting the next message.
 
 Error Reporting
 ---------------

--- a/docs/source/global.rst
+++ b/docs/source/global.rst
@@ -1,6 +1,7 @@
 .. References
 
 .. |AgeLimit| replace:: :class:`AgeLimit<dramatiq.middleware.AgeLimit>`
+.. |AsyncIO| replace:: :class:`AsyncIO<dramatiq.middleware.AsyncIO>`
 .. |Barriers| replace:: :class:`Barriers<dramatiq.rate_limits.Barrier>`
 .. |Brokers| replace:: :class:`Brokers<dramatiq.Broker>`
 .. |Broker| replace:: :class:`Broker<dramatiq.Broker>`

